ec2utils/ec2uploadimg/lib/ec2utils/ec2uploadimg.py

```diff
@@ -696,15 +696,25 @@ def _register_image(self, snapshot):
 
         root_device_name = self._determine_root_device()
         if self.image_virt_type == 'hvm':
-            ami = self._connect().register_image(
-                Name=self.image_name,
-                Description=self.image_description,
-                Architecture=self.image_arch,
-                RootDeviceName=root_device_name,
-                BlockDeviceMappings=block_device_map,
-                VirtualizationType=self.image_virt_type,
-                SriovNetSupport=self.sriov_type
-            )
+            if self.sriov_type:
+                ami = self._connect().register_image(
+                    Name=self.image_name,
+                    Description=self.image_description,
+                    Architecture=self.image_arch,
+                    RootDeviceName=root_device_name,
+                    BlockDeviceMappings=block_device_map,
+                    VirtualizationType=self.image_virt_type,
+                    SriovNetSupport=self.sriov_type
+                )
+            else:
+                ami = self._connect().register_image(
+                    Name=self.image_name,
+                    Description=self.image_description,
+                    Architecture=self.image_arch,
+                    RootDeviceName=root_device_name,
+                    BlockDeviceMappings=block_device_map,
+                    VirtualizationType=self.image_virt_type
+                )
```

Comment:
Rather than this implementation I think it would be construct a dictionary that is then used as kwargs dict, i.e. passed with **kwargs and populate the dictionary accordingly. Constructing the kwargs allows us not to have "sub-types" of the registration call for HVM images. We could also expand this concept to the PV images and thus only end up with one self._connect().register_image(...) call that simply receives the kwargs dict.